nemo_curator/modules/exact_dedup.py

```diff
@@ -166,10 +173,38 @@ def __call__(self, dataset: DocumentDataset) -> Union[DocumentDataset, str]:
         self._logger.info(
             f"Time taken for Exact Dedup Computation = {time.time() - t0}s and output written at {write_path}"
         )
-        if is_cudf_type(result):
-            import dask_cudf
+        backend = "cudf" if is_cudf_type(result) else "pandas"
+        return DocumentDataset.read_parquet(
```

Comment:
Given that the result here is already shuffled on the hash key which cannot be avoided. Do you think it's worth leveraging that for the removal step, or do you prefer have custom removal logic at the cost of shuffling again?